scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,355 @@
+                  images: res.result ?? [],
+                });
+              } catch {
+                this.setState({
+                  selectedImage: null,
+                  images: [],
+                });
+              }
+            }
+          })
+          .catch(() => {
+            this.setState({ fetchError: true });
+          })
+          .finally(() => this._mounted && this.setState({ isLoading: false }));
+      });
+    }
+
+    isEqualTags = (current: Array<ITagInput>, prev: Array<ITagInput>) =>
+      JSON.stringify(current) === JSON.stringify(prev)
+
+    formatTags = (concepts: IServerResponse) => {
+      let res: Array<ITagInput> = [];
+      for (const key in concepts) {
+        if (key === "subject") {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: "category",
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        } else if (
+          key === "organisation" ||
+          key === "person" ||
+          key === "event" ||
+          key === "place" ||
+          key === "object"
+        ) {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: key,
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        }
+      }
+      return res;
+    }
+
+    handleClickImage = (image: IImage) => {
+      this.setState({ selectedImage: image });
+    }
+
+    handleDragStart = (
+      event: React.DragEvent<HTMLDivElement>,
+      image: IImage | null,
+    ) => {
+      if (!image) {
+        return;
+      }
+      let item: IArticle;
+      try {
+        item = {
+          ...this.props.article,
+          type: "picture",
+          description_text: image?.caption ?? "",
+          headline: image?.headline ?? "",
+          source: image?.source ?? "",
+          fetch_endpoint: "scanpix",
+          renditions: {
+            thumbnail: {
+              href: image?.thumbnailUrl ?? "",
+              mimetype: "image/jpeg",
+            },
+            viewImage: {
+              href: image?.imageUrl ?? "",
+              mimetype: "image/jpeg",
+            },
+            baseImage: {
+              href: image?.imageUrl ?? "",
+              mimetype: "image/jpeg",
+            },
+          },
+          byline: image?.byline ?? "",
+        };
+      } catch {
+        return;
+      }
+      event.dataTransfer.setData(
+        "application/superdesk.item.picture",
+        JSON.stringify(item),
+      );
+    }
+
+    render() {
+      const { style } = this.props;
+      const { showImages, isLoading, fetchError, selectedImage, images } =
+        this.state;
+
+      if (!this._mounted) {
+        return null;
+      }
+      return (
+        <ToggleBoxNext
+          title={`image suggestions ${
```

Comment:
Noted, thanks!

Reply:
I was about to comment that usage of `gettext` is incorrect - you can't interpolate a string with JS and have to use `{{placeholder}}` syntax inside the string. On the other hand it would probably be better not to store `fetchError` in the state in the first place if the only purpose is to display it in the title. Better use `superdesk.ui.alert` to show an alert and suggest user to retry the action.